### ISSUE TITLE
uniqueItems breaks an array of strings

### ISSUE THREAD

**Issue by odscjames** [json-editor/json-editor#735]
### General information

- json-editor version: 2.2.1

Seen in Chrome and FireFox

### Expected behavior

I can edit an array of strings  with uniqueItems set

### Actual behavior

Section is blank on form

![Screenshot from 2020-04-28 17-08-45](https://user-images.githubusercontent.com/37832009/80510758-12549980-8973-11ea-8f25-3b98dc609afa.png)

In the output, any existing data is removed

### Steps to reproduce the behavior

Direct Link: https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgAJEtXqUIZVCgQUAelMda8ALQ61ZAHTSA5qYAmUskSjWADABZTO1kAYgU3GgwpGGg4eEIFWChsKkIAeSlnMngYRDJYBAACGAj4WCIYKilC3EQoQoilaNiERMEIVP0wTm5xNggpSCrYGg0QAXhnETJnUbQQJJTCABlsqZnRtkaomIL4/QB1ZUxlGrXp2cRiq6hjq9qoFkLst1NpQtuqYrKqojIMGgfMDPDAAxBXGCPZ4ANzIMAEeBS3wA/G0OoQyFJPIIFJC1Mh5lB2p1FFBopMFHQYPBllRJrdUABGNiLEkASQAymlCgAmaxuGDOSFnSYXL5YCIgAC+bBEOQAjiIqGyoPjUGSlVKtWxEMowBReFVBlICSBqarPDw4oldCS8GAwFA6p4IP5pVKgA

If you remove "uniqueItems": true it works great.

SCHEMA:

```javascript
{
    "$schema": "http://json-schema.org/draft-04/schema#", 
    "description": "", 
    "title": "Organisation identifier list description", 
    "type": "object", 
    "properties": {
       



                "languages": {
                    "title": "Languages", 
                    "description": "Which languages is this list, and/or the interfaces to access it, available in?", 

                    "type": "array", 
                    "items": {
                        "type": "string",
                        "minLength": 1, 
                        "title": "ISO 2-digit language code" 
                    }, 
                    "uniqueItems": true
                }
            
        

    }
}
```

DATA:

```javascript

{

    "languages": [
      "en"
    ]
}
```

**Comment by mattjenny** [comment]
I think the issue is due to this block: https://github.com/json-editor/json-editor/blob/master/src/resolvers.js#L67-L72, which is overriding the format to "multiselect" in this case. If I understand that block correctly, I think that logic should only be applied if the strings in the array have `enum` values defined; otherwise, the proper type for the component should be an array.

**Comment by schmunk42** [comment]
A PR with tests would be highly appreciated.

**Comment by germanbisurgi** [comment]
i don't think that the combination "uniqueItems" and "enum" was intended to be edited. I believe they are used in combination to give the user an options list to choose for and not to be able to add or remove options. It should not be editable. t looks to me that what you need is a custom validator that proves if the values in an editable array (add / remove) are unique. But i could misunderstood the problem

**Comment by germanbisurgi** [comment]
fixed in #843
